src/client/pydaos/pydaos_shim.c

```diff
@@ -222,12 +222,23 @@ cont_open(int ret, char *pool, char *cont, int flags)
 	if (prop)
 		daos_prop_free(prop);
 	if (rc) {
-		if (daos_handle_is_valid(oh))
-			daos_kv_close(oh, NULL);
-		if (daos_handle_is_valid(coh))
-			daos_cont_close(coh, NULL);
-		if (daos_handle_is_valid(poh))
-			daos_pool_disconnect(poh, NULL);
+		int	rc2;
+
+		if (daos_handle_is_valid(oh)) {
+			rc2 = daos_kv_close(oh, NULL);
+			if (rc2)
+				D_ERROR("failed to close kv\n");
```

Comment:
Should you print the value of rc2?

Reply:
we could but I don't think it matters much here since the actual error should be reported above. if a close fails that probably can lead to more confusion?